cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -69,9 +70,10 @@ func CreateAwsManager(configReader io.Reader) (*AwsManager, error) {
 
 	service := autoscaling.New(session.New())
 	manager := &AwsManager{
-		asgs:     make([]*asgInformation, 0),
-		service:  service,
-		asgCache: make(map[AwsRef]*Asg),
+		asgs:                     make([]*asgInformation, 0),
+		service:                  service,
+		asgCache:                 make(map[AwsRef]*Asg),
+		instancesNotInManagedAsg: make(map[AwsRef]struct{}),
```

Comment:
> And if you'd just like to regenerateCache every X seconds, theoretically, we can just do it by checking if elapsed time since the last regeneration is greater than X, in the very beginning of regenerateCache?

sounds good to me